pkg/rpcclient/wsclient.go

```diff
@@ -559,3 +559,82 @@ func (c *WSClient) GetError() error {
 	}
 	return c.closeErr
 }
+
+// Wait waits until transaction is accepted to the chain and returns its execution result
+// or an error if it's missing from chain after VUB block.
+func (c *WSClient) Wait(h util.Uint256, vub uint32) (res *state.AppExecResult, waitErr error) {
+	var wsWaitErr error
+	defer func() {
+		if wsWaitErr != nil {
+			res, waitErr = c.Client.Wait(h, vub)
+			if waitErr != nil {
+				waitErr = fmt.Errorf("WS client error: %w, client error: %v", wsWaitErr, waitErr)
+			}
+		}
+	}()
+	rcvr := make(chan Notification)
+	defer func() {
+	drainLoop:
+		// Drain rcvr to avoid other notification receivers blocking.
+		for {
+			select {
+			case <-rcvr:
+			default:
+				break drainLoop
+			}
+		}
+		close(rcvr)
+	}()
+	blocksID, err := c.subscribeForNewBlocksInternal(nil, rcvr)
```

Comment:
`since` filter is added. `headersOnly` filter is a bit more complicated, to support it on the server side we firstly need to iterate through the whole set of filters of the given subscriber and decide whether there's at least one subscription without `headersOnly` filter. Only if such subscriptions are missing we can marshal header. Do we need to implement it?

Another solution is to create a separate "header_added" notification event.

Reply:
Let's create an issue for header notifications, it seems to be useful to me.

Reply:
https://github.com/nspcc-dev/neo-go/issues/2751